subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/internal/AbstractCodeQualityPlugin.java

```diff
@@ -229,4 +233,10 @@ protected void withBasePlugin(Action<Plugin> action) {
     protected JavaPluginExtension getJavaPluginExtension() {
         return project.getExtensions().getByType(JavaPluginExtension.class);
     }
+
+    public static void maybeAddOpensJvmArgs(JavaLauncher javaLauncher, ForkingWorkerSpec spec) {
+        if (JavaVersion.toVersion(javaLauncher.getMetadata().getJavaRuntimeVersion()).isJava9Compatible()) {
+            spec.getForkOptions().jvmArgs("--add-opens", OPEN_MODULES_ARG);
```

Comment:
Because down in the guts of one of the classes `IsolatedAntBuilder` uses, it does [this](https://github.com/gradle/gradle/blob/9a2009ab1cb9ffb347a47c4a9cea16de559a7d82/subprojects/core/src/main/java/org/gradle/api/internal/classloading/PreferenceCleaningGroovySystemLoader.java#L34) which throws a [module access error](https://ge.gradle.org/s/fompyhqajkxpq/tests/:code-quality:embeddedIntegTest/org.gradle.api.plugins.quality.checkstyle.CheckstylePluginVersionIntegrationTest/analyze%20good%20code%207.0?focused-exception-line=0-1-2-3-4-5-6-7-8-0&top-execution=1).